src/sources/kafka.rs

```diff
@@ -179,6 +183,21 @@ fn kafka_source(
                                 log.insert(offset_key, Value::from(msg.offset()));
                             }
 
+                            if let Some(headers_key) = &headers_key {
+                                let mut headers_map = BTreeMap::new();
+                                if let Some(headers) = msg.headers() {
+                                    for i in 0..headers.count() {
+                                        let header =
+                                            headers.get(i).expect("header index is out of bounds");
+                                        headers_map.insert(
+                                            header.0.to_string(),
+                                            Bytes::from(header.1.to_owned()).into(),
+                                        );
+                                    }
+                                }
```

Comment:
This would probably be easier done with iterators and adapters (if `msg.headers()` returns an owned vec and not a reference, used `into_iter()`):
```suggestion
                                let headers_map = msg
                                    .headers()
                                    .map(|headers| {
                                        headers
                                            .iter()
                                            .map(|(key, value)| (key.to_string(), Bytes::from(value.to_owned()).into()))
                                            .collect()
                                    })
                                    .unwrap_or_else(|| BTreeMap::default());
```

Reply:
Agreed. `.headers()` actually returns a trait object: https://docs.rs/rdkafka/0.26.0/rdkafka/message/trait.Headers.html . I was a little surprised to find this.

Reply:
@bruceg , @jszwedko  - I can add comment below this `for` loop mentioning this is because `Headers` do not proivde `.iter`.